decidim-budgets/spec/system/admin_manages_budgets_spec.rb

```diff
@@ -111,6 +111,15 @@
     end
   end
 
+  describe "when managing a budget with scopes" do
+    let!(:budget) { create(:budget, component: current_component) }
+    let!(:scope) { create(:scope, organization: current_component.organization) }
```

Comment:
As you'®e not actually assigning the scope to anything this test will always pass, no matter the code.

When I do a `have_no_content` I think its a good practice to do the reverse context (`have_content`) just to be sure that this is actually doing what we want to do. 

This is what I had in mind, but mind that I couldn't get it work yet 

```diff
diff --git a/decidim-budgets/spec/system/admin_manages_budgets_spec.rb b/decidim-budgets/spec/system/admin_manages_budgets_spec.rb
index 5f6cd4d9f9..7a9328f9e5 100644
--- a/decidim-budgets/spec/system/admin_manages_budgets_spec.rb
+++ b/decidim-budgets/spec/system/admin_manages_budgets_spec.rb
@@ -7,6 +7,7 @@ describe "Admin manages budgets" do
   let(:manifest_name) { "budgets" }

   include_context "when managing a component as an admin"
+
   before do
     budget
     switch_to_host(organization.host)
@@ -111,6 +112,35 @@ describe "Admin manages budgets" do
     end
   end

+  describe "managing budgets" do
+    let(:scopes) { create_list(:scope, 3, organization: organization) }
+    let!(:component) do
+      create(:component, manifest:, settings: { scopes_enabled: }, participatory_space:)
+    end
+
+    before do
+      visit current_path
+    end
+
+    context "when the scope has subscopes" do
+      let(:scopes_enabled) { true }
+
+      it "displays the scope column" do
+        expect(component.has_subscopes?).to be true
+        expect(page).to have_content("Scope")
+      end
+    end
+
+    context "when the scope does not have subscopes" do
+      let(:scopes_enabled) { false }
+
+      it "does not display the scope column" do
+        expect(component.has_subscopes?).to be false
+        expect(page).to have_no_content("Scope")
+      end
+    end
+  end
+
   describe "component page shows finished and pending orders of all budgets" do
     context "when component has many budgets with orders" do
       let(:budget2) { create(:budget, :with_projects, component: current_component) }
```

Maybe someone smarter than me can fix this spec 😄 ? @alecslupu would you like to give it a spin?

Reply:
Understood. If @alecslupu has a suggestion to improve the test range, I am all ears.

Reply:
TBH, i would use: 

```ruby
describe "when managing a budget with scopes" do
    let!(:scopes) { create_list(:subscope, 3, organization: organization, parent: scope) }
    let(:scope_id) { scope.id }
    let(:participatory_space) { create(:participatory_process, organization: organization, scopes_enabled: ) }
    let!(:component) { create(:component, manifest:, settings: { scopes_enabled:, scope_id: }, participatory_space:) }
    let!(:budget) { create(:budget, component: current_component) }
    let!(:scope) { create(:scope, organization: organization) }
    let(:scopes_enabled) { true }

    before do
      visit current_path
    end

    context "when the scope has subscopes" do
      context "when scopes_enabled is true" do
        it "displays the scope column" do
          expect(component.scopes_enabled?).to be_truthy
          expect(component.has_subscopes?).to be_truthy
          expect(page).to have_content("Scope")
        end
      end

      context "when scopes_enabled is false" do
        let(:scopes_enabled) { false }

        it "displays the scope column" do
          expect(component.scopes_enabled?).to be_falsey
          expect(component.has_subscopes?).to be_falsey
          expect(page).not_to have_content("Scope")
        end
      end
    end

    context "when the scope does not have subscopes" do
      let(:scope_id) { scopes.first.id }

      context "when scopes_enabled is true" do
        it "hides the scope column" do
          expect(component.scopes_enabled?).to be_truthy
          expect(component.has_subscopes?).to be_falsey
          expect(page).not_to have_content("Scope")
        end
      end

      context "when scopes_enabled is false" do
        let(:scopes_enabled) { false }

        it "displays the scope column" do
          expect(component.scopes_enabled?).to be_falsey
          expect(component.has_subscopes?).to be_falsey
          expect(page).not_to have_content("Scope")
        end
      end
    end
  end

```

Reply:
Thanks very much @alecslupu will update the test suite and request a review!